src/python/pants/core/goals/test.py

```diff
@@ -188,15 +191,14 @@ def path_safe_description(self) -> str:
         return f"{self.addresses[0].path_safe_spec}+{len(self.addresses)-1}"
 
     def __lt__(self, other: Any) -> bool:
-        """We sort first by status (skipped vs failed vs succeeded), then alphanumerically within
-        each group."""
+        """We sort first by exit code, then alphanumerically within each group."""
```

Comment:
For a future change: I don't actually think we want to sort by exit_code but just by success/failure. If different tests have different exit codes, we want them to be a single bucket of failed tests that is sorted alphabetically by description, not multiple buckets.